### PR TITLE
feat(deps): bump minimum @typescript-eslint to v7.0.0

### LINKED ISSUE #1453: Update @typescript-eslint/eslint-plugin
`npm ERR!   dev @typescript-eslint/eslint-plugin@"^7.1.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @typescript-eslint/eslint-plugin@"^6.4.0" from eslint-config-standard-with-typescript@43.0.1
npm ERR! node_modules/eslint-config-standard-with-typescript
npm ERR!   dev eslint-config-standard-with-typescript@"^43.0.1" from the root project`

### PR DESCRIPTION
**Pull request by mihaerzen**
**What is the purpose of this pull request? (put an "X" next to item)**

- [x] Documentation update
- [ ] Bug fix
- [ ] New feature
- [x] Other, please explain:

Hi all! 👋🏼 

I was annoyed by the outdated `@typescript-eslint/parser` and I really wanted to use this amazing set of eslint rules, so I went ahead and updated it.

This is my first PR to this project, so please do let me know if I made some apparent mistakes. Thank you for maintaining this project! 🙇🏼 

**What changes did you make? (Give an overview)**

I have updated the `@typescript-eslint/parser`. I had to also update the `@typescript-eslint/eslint-plugin`.

**Which issue (if any) does this pull request address?**

Fixes #1453
Closes #1431

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -227,6 +228,11 @@ const config = {
     '@typescript-eslint/triple-slash-reference': ['error', { lib: 'never', path: 'never', types: 'never' }],
     '@typescript-eslint/type-annotation-spacing': 'error',
     '@typescript-eslint/unbound-method': ['error', { ignoreStatic: false }],
+    '@typescript-eslint/no-array-delete': 'error',
+    '@typescript-eslint/no-unsafe-unary-minus': 'error',
+    '@typescript-eslint/no-useless-template-literals': 'error',
+    '@typescript-eslint/prefer-find': 'error',
+    '@typescript-eslint/use-unknown-in-catch-callback-variable': 'error',
```
Comment (mightyiam) [inline src/index.ts:235]: If you could separate adding rules into a different PR, I would prefer that, please.

Comment (mihaerzen) [reply src/index.ts:235]: I don't think I can separate upgrading `@typescript-eslint/parser` (and consecutively also `@typescript-eslint/eslint-plugin`) and introduced rules into separate PR's. These rules are a part of `@typescript-eslint/eslint-plugin` and tests started failing because of that upgrade so I had to add these. I am also not sure if I can introduce these rules prior upgrading (meaning I would add these rules one by one and then upgrade), and honestly it feels like more work for the same result. 🤷‍♂️

Comment (mightyiam) [reply src/index.ts:235]: Sorry. Perhaps the code is not self-explanatory enough.

In the test file there's a list of rules that are not yet considered. Rules that are new in the plugin should probably get added to that list when bumping the plugin.

Comment (mihaerzen) [reply src/index.ts]: Gotcha. I have updated the code :v:

Comment (mihaerzen) [reply src/index.ts]: `prefer-promise-reject-errors` is an "equivalent", however. Not sure how to skip it. The test explicitly tests for it to be there so I would say it should be there. Unless the test is wrong? Let me know or just approve the PR.

Comment (mihaerzen) [reply src/index.ts]: @mightyiam is there anything else I should to do get this merged?

**Comment by mightyiam** [comment]
I feel that there's some stuff that should occur before this. For one, the absorbing of eslint-config-standard. Which requires some test changes. I have quite a lot of time tomorrow and Thursday...

**Comment by mihaerzen** [comment]
> absorbing of eslint-config-standard

My apologies, I do not exactly understand what you mean by this. Can you elaborate what you mean with this exactly and how is this related to this PR so I can fix it. Thank you 🙏🏼

**Comment by mightyiam** [comment]
It's related because it simplifies the testing.

But it's not strictly necessary. I'm sure your PR can be brought through without it.

I'll take another look.

**Comment by matthargett** [comment]
A quick +1 on the intent of the PR. When using `nx` and migrating to its latest version (18.2.0), we'd (seemingly) have to abandon eslint-config-love to do so. It looks like merging this PR (and publishing a new release) would help get us unblocked.
```
npm ERR! Found: @typescript-eslint/eslint-plugin@7.4.0
npm ERR! node_modules/@typescript-eslint/eslint-plugin
npm ERR!   dev @typescript-eslint/eslint-plugin@"7.4.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @typescript-eslint/eslint-plugin@"^6.4.0" from eslint-config-love@43.1.0
npm ERR! node_modules/eslint-config-love
npm ERR!   dev eslint-config-love@"^43.1.0" from the root project
```

Happy to test an -rc or similar package, if that's helpful!

**Comment by mightyiam** [comment]
Sorry for breaking your PR, @mihaerzen ! 

It was broken with #1483 which was a precursor for #1484 which provides the test coverage needed in order to safely absorb eslint-config-standard.

If you're upset and want me to bump typescript-eslint, please let me know.

**Comment by mihaerzen** [comment]
> If you're upset and want me to bump typescript-eslint, please let me know.

Not a problem, happy to rebase. Seeing `resolved-config.ts` isn't this making the `standard.ts` and `equivalents.ts` (maybe also some others) redundant? Personally I feel `resolved-config.ts` is a much better test as it covers a full end result and it could be considered the "source of truth".

I have rebased and with the necessary upgrade of `typescript-eslint` plugin, the existing `prefer-promise-reject-errors` according to [how to use section](https://typescript-eslint.io/rules/prefer-promise-reject-errors/#how-to-use) should be disabled.

Let me know what you think.

Cheers
✌️

**Review by mightyiam — COMMENTED** [review]
And the commit message

> feat(deps): bump minimum @typescript-eslint to v7.0.0
>
> BREAKING CHANGE: minimum @typescript-eslint is v7.0.0

Thank you

**Comment by mightyiam** [comment]
Thank you for revising.

I found that v7.0.0 had a peerDep related bug. So we're bumping to v7.0.1 in which that seems to have been fixed.

I found that the `package-lock.json` had seemingly unrelated changes. So I re-did the lock change.

I also set the bottom versions to v7.0.1 (they were v7.4.0).

**Comment by mightyiam** [comment]
Thank you, @mihaerzen .

**Comment by mihaerzen** [comment]
@mightyiam Thank you for your patience ✌️
